go/host/host.go

```diff
@@ -716,6 +699,47 @@ func (a *Node) monitorBlocks() {
 	subs.Unsubscribe()
 }
 
+func (a *Node) catchupMissedBlocks(lastKnownBlkHash gethcommon.Hash) {
+	var lastBlkNumber *big.Int
+	var reingestBlocks []*types.Block
+
+	// get the tip block
+	lastBlk, err := a.ethClient.BlockByNumber(lastBlkNumber)
+	if err != nil {
+		log.Panic("catching up on missed blocks, unable to fetch tip block - reason: %s", err)
+	}
+
+	if lastBlk.Hash().Hex() == lastKnownBlkHash.Hex() {
+		// no new blocks have been issued
+		return
+	}
+	reingestBlocks = append(reingestBlocks, lastBlk)
+
+	// iterate from the tip (last block produced) to the last block ingested by the node
+	for lastBlk.Hash().Hex() != lastKnownBlkHash.Hex() {
```

Comment:
I think one of these should be called latestBlock maybe? I guess one of them is the latest block we have received from the L1 but it's not clear to me which it is. Some renaming or comment to clarify perhaps (sorry, Friday afternoon Matt is not the sharpest, it's probably not as confusing as I'm making out)